polyfill/lib/ecmascript.mjs

```diff
@@ -3440,20 +3404,21 @@ export const ES = ObjectAssign({}, ES2020, {
 
     // First convert time units up to days, if rounding to days or higher units.
     // If rounding relative to a ZonedDateTime, then some days may not be 24h.
+    let dayLengthNs;
     if (unit === 'years' || unit === 'months' || unit === 'weeks' || unit === 'days') {
       nanoseconds = ES.TotalDurationNanoseconds(0, hours, minutes, seconds, milliseconds, microseconds, nanoseconds, 0);
```

Comment:
The existing code in this method, like much of the polyfill, often re-uses existing variables instead of defining new ones. In the case of `nanoseconds`, this (existing) code changes the type of `nanoseconds` from Number to a BigInt before changing it back to a Number lower down in the method. Is this OK?  Or should I refactor to prevent type-changing re-use of variables like this?

Reply:
I think it's fine, but then again I did write it.